pgjdbc/src/main/java/org/postgresql/jdbc/PgBlobBytea.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2004, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+/*
+ * Implementation of Blob interface backed by Postgres bytea fields.
+ *
+ * See PgBlob.java for implementation using Postgres Large Objects.
+ */
+
+package org.postgresql.jdbc;
+
+import org.postgresql.util.GT;
+import org.postgresql.util.PSQLException;
+import org.postgresql.util.PSQLState;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.sql.Blob;
+import java.sql.SQLException;
+import java.util.Arrays;
+
+public class PgBlobBytea implements java.sql.Blob {
+
+  private byte[] data;
+
+  public static final int MAX_BYTES = 1073741824; // 1GiB max size of a bytea
+
+  public PgBlobBytea() {
+    this.data = new byte[0];
```

Comment:
I think setting this to null makes more sense as it does not allocate an object. The byte[0] is never used anywhere

Reply:
I started out using null, but the Checker complained loudly, so I got rid of it almost completely in commit c8c1bbf1. I'll be happy to revert that, I agree it's more natural, if you can tell me how to keep the checker happy.

Reply:
Yes, revert that, we will fix checker. At worst you can tell it to ignore it

Reply:
There's nothing wrong with using `byte @Nullable [] data`
Please refer to https://github.com/pgjdbc/pgjdbc/blob/master/CONTRIBUTING.md#null-safety